### ISSUE TITLE
When executing the ampl interface with "halt_on_ampl_error yes" the rest of the script bails out.

### ISSUE THREAD

**Issue by svigerske** [coin-or/Ipopt#10]
Issue created by migration from Trac.

Original creator: soyrush@gmail.com

Original creation time: 2006-05-16 02:36:55

Assignee: ipopt-team

Version: 3.0

I'm not sure if this is the correct way that Ipopt should work, or if it's a bug. When I execute my ampl script with the "halt_on_ampl_error yes" option I get the following error message:

Error evaluating "var =" definition 2170:
can't evaluate exp(2998.13): No such file or directory.
exit code 1
<BREAK>

After this error, the rest of my script is not executed, and therefore it becomes quite hard to find where or which variable is the problem (i.e., where to set a new bound). Specially because after the error the values of the variables in AMPL don't get updated...

If I remove the halt_on... option the solver continues, but doesn't converge probably due to the incorrect bound.

I'm working on a Linux machine with the following characteristics:
Linux distro: CentOS 4.3(Final)
gcc version 3.4.5
I compiled my ipopt with the default options obtainded by executing the configure script, and with ma27, ma19.
ASL, blas, and Lapack where donwloaded with the get* script.

Thanks all,
 Rodrigo López Negrete

**Comment by svigerske** [comment]
Comment by claird@andrew.cmu.edu created at 2006-05-16 14:33:41

When Ampl has difficulty evaluating an expression in a constraint, like exp(2998.13), it will either halt and report the error or return a Nan or Inf to Ipopt and continue. The option "halt_on_ampl_error" is present to decide which behavior is preferred. When you set "halt_on_ampl_error yes", execution will stop when there is an error and you see the AMPL error message to help find the problem ("Error evaluating "var =" definition 2170: can't evaluate exp(2998.13)"). If you set "halt_on_ampl_error no" (or leave it at the default of no), then Ipopt will try and continue (by cutting back the step, for instance).

In summary, the behavior you observe is exactly as the option was designed. Please let us know if there is a suggestion for an alternative behavior.

Cheers,

Carl.

**Comment by svigerske** [comment]
Comment by claird@andrew.cmu.edu created at 2006-05-16 14:33:41

Changing keywords from  to halt_on_ampl_error,error evaluating".

**Comment by svigerske** [comment]
Comment by claird@andrew.cmu.edu created at 2006-05-16 14:33:41

Resolution: wontfix
